include/solo5/mft_abi.h

```diff
@@ -99,55 +100,76 @@ struct mft {
 };
 
 /*
- * Manifest ELF note headers.
+ * Maximum supported number of manifest entires.
+ */
+#define MFT_MAX_ENTRIES 64
+
+/*
+ * HERE BE DRAGONS.
  *
- * Note that somewhat counter-intuitively, in practice the note section header
- * fields (namesz/descsz/type) are 32-bit words even on ELF64.
+ * The following structures and macros are used to declare a Solo5 "MFT1"
+ * format note at link time. This is somewhat tricky, as we need to ensure all
+ * structures are aligned correctly.
  */
-#define SOLO5_NOTE_NAME     "Solo5"
-#define SOLO5_NOTE_MANIFEST 0x3154464d /* "MFT1" */
+#define SOLO5_NOTE_NAME      "Solo5"
+#define SOLO5_NOTE_TYPE_MFT1 0x3154464d /* "MFT1" */
 
-struct mft_note_header {
-    uint32_t namesz;
-    uint32_t descsz;
-    uint32_t type;
-    char name[(sizeof(SOLO5_NOTE_NAME) + 3) & -4];
+/*
+ * Defines an Elf64_Nhdr with n_name filled in and padded to a 4-byte boundary,
+ * i.e. the common part of a Solo5-owned Nhdr.
+ */
+struct mft1_nhdr {
+    uint32_t n_namesz;
+    uint32_t n_descsz;
+    uint32_t n_type;
+    char n_name[(sizeof(SOLO5_NOTE_NAME) + 3) & -4];
 };
 
-struct mft_note {
-    struct mft_note_header h;
+/*
+ * Defines the entire note (header, descriptor content).
+ */
+struct mft1_note {
+    struct mft1_nhdr h;
+    /*
+     * (m) will be aligned to a MFT_NOTE_ALIGN boundary.
+     */
     struct mft m;
```

Comment:
Are we sure that we never want to change the `mft` format, i.e. should this also be `struct mft1`?

Reply:
> Would it make sense to make this MFT1_MAX_ENTRIES ?

Related:

> Are we sure that we never want to change the mft format, i.e. should this also be struct mft1?

I'm not sure. `struct mft` has its own inner versioning, and I'm hoping that we don't need to change it in an incompatible way. I'm also lazy to go changing `mft` to `mft1` throughout the codebase. But if you think it's worth it, I'll do that. Let me know.

Reply:
I found it a bit odd that some of them dealing with MFTv1 were versioned and some not, but I think your rationale makes sense.

I think the general MFT structure seems extensible enough (`struct mft` with version, count, `mft_entry` VLA) - for some applications it might make sense to be able to adjust the `MAX_MFT_ENTRIES` constant, but that seems like a backwards compatible change going forward too, if need be.

To sum up, I think `struct mft_entry` seems like the most likely candidate for changes. Not sure either if it's worth changing everything for that at the moment.

Reply:
> To sum up, I think struct mft_entry seems like the most likely candidate for changes. Not sure either if it's worth changing everything for that at the moment.

I agree. I do want to make some small changes to `struct mft_entry`, to allow for storing something other than a `hostfd`. Will do so in a separate PR, as that doesn't directly concern the NOTE loading.